crates/sui-core/src/transaction_driver/mod.rs

```diff
@@ -88,41 +96,147 @@ where
             client_monitor,
         });
 
+        let driver_clone = driver.clone();
+
+        spawn_logged_monitored_task!(Self::run_latency_checks(driver_clone));
+
         driver.enable_reconfig(reconfig_observer);
         driver
     }
 
-    #[instrument(level = "error", skip_all, fields(tx_digest = ?request.transaction.digest()))]
+    // Runs a background task to send ping transactions to all validators to perform latency checks to test both the fast path and the consensus path.
+    async fn run_latency_checks(self: Arc<Self>) {
+        const MAX_DELAY_BETWEEN_REQUESTS_MS: u64 = 5_000;
+        const INTERVAL_BETWEEN_RUNS: Duration = Duration::from_millis(10_000);
+        const PING_REQUEST_TIMEOUT: Duration = Duration::from_millis(5_000);
+        let mut interval = interval(INTERVAL_BETWEEN_RUNS);
+        interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+
+        loop {
+            interval.tick().await;
+
+            // We are iterating over the single writer and shared object transaction types to test both the fast path and the consensus path.
+            let mut tasks = JoinSet::new();
+
+            let auth_agg = self.authority_aggregator.load().clone();
+
+            // Get the validators by their total score
+            let mut clients_by_total_score = self
+                .client_monitor
+                .validators_by_total_score(&auth_agg.committee);
+
+            // Order the clients by their total score in score descending order
+            clients_by_total_score.sort_by(|a, b| b.1.partial_cmp(&a.1).unwrap());
+
+            // Now keep only the clients after the first K as those being used on the retrier, and at least half of the committee.
+            let k = (auth_agg.committee.num_members() / TOP_K_VALIDATORS_DENOMINATOR)
+                .max(auth_agg.committee.num_members() / 2);
```

Comment:
Or `auth_agg.committee.num_members() / TOP_K_VALIDATORS_DENOMINATOR.min(2)`. Or how about just increasing interval by 2x or 3x?

Reply:
Yeah most probably I'll increase the interval. It simplifies things and makes less assumptions.